### DISCUSSION TITLE
Request for Moirai Fine-Tuning Example

### DISCUSSION THREAD

**Discussion by AhanBhatt** [SalesforceAIResearch/uni2ts#148] [Q&A]
Hi,  

Love the work you guys are doing! I’m trying to fine-tune it for a project but can’t figure out how to do it without the CLI. Could you share a Jupyter notebook or Python script example for fine-tuning?  

Thanks!

**Comment by liyujia011025** [comment]
Hello expert:
I am currently trying to fine tune Moirai using my own dataset, but I have found that the effect has actually worsened after fine-tuning. May I ask if you have resolved the issue with fine-tuning? Have you tried changing some hyperparameters during the fine-tuning process to make the fine-tuning operation more effective?
Looking forward to your reply!

**Reply by AhanBhatt** [reply]
Hey, I'm not really an expert but I did try fine-tuning Moirai. Unfortunately, I was not able to fine-tune the model as the documented method for fine-tuning the model (CLI) gave me a lot of errors. I got acceptableresults via zero-shot forecasting and decided to go ahead with that. Sorry I couldn't be of any help. Hope you figure it out!

**Comment by chenghaoliu89** [comment]
Hi @AhanBhatt @liyujia011025 thanks for your interest. We are working on the enhanced fine-tuning module of uni2ts. @zqiao11 has submitted a PR #189. Let me know if you have any further suggestion/requirements for better fine tuning guide line. If you face any issue with your fine-tuning data and evaluation data. You can share it here.

**Reply by liyujia011025** [reply]
> Hi @AhanBhatt @liyujia011025 thanks for your interest. We are working on the enhanced fine-tuning module of uni2ts. @zqiao11 has submitted a PR #189. Let me know if you have any further suggestion/requirements for better fine tuning guide line. If you face any issue with your fine-tuning data and evaluation data. You can share it here.

Thank you very much for your reply and your efforts in the fine-tuning work.
But I have found that using datasets that the model has never seen before, such as the wind power forecasting dataset, fine-tuning Moirai may not achieve good results, which may also be due to the intermittency and instability of wind power itself. Due to the current popularity of research on renewable energy forecasting, if possible, could you please provide an example of fine-tuning using wind power datasets (like publicly available datasets such as Global Energy Forecasting Competition 2012 and Global Energy Forecasting Competition 2014)? I would greatly appreciate it!
https://www.sciencedirect.com/science/article/abs/pii/S0169207013000745
https://www.sciencedirect.com/science/article/abs/pii/S0169207015001521

**Comment by liyujia011025** [comment]
Thank you very much for your reply and your efforts in the fine-tuning work.
But I have found that using datasets that the model has never seen before, such as the wind power forecasting dataset, fine-tuning Moirai may not achieve good results, which may also be due to the intermittency and instability of wind power itself. Due to the current popularity of research on renewable energy forecasting, if possible, could you please provide an example of fine-tuning using wind power datasets (like publicly available datasets such as Global Energy Forecasting Competition 2012 and Global Energy Forecasting Competition 2014)? I would greatly appreciate it!
https://www.sciencedirect.com/science/article/abs/pii/S0169207013000745
https://www.sciencedirect.com/science/article/abs/pii/S0169207015001521




---- Replied Message ----
| From | Chenghao ***@***.***> |
| Date | 2/18/2025 22:46 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [SalesforceAIResearch/uni2ts] Request for Moirai Fine-Tuning Example (Discussion #148) |

Hi @***@***.*** thanks for your interest. We are working on the enhanced fine-tuning module of uni2ts. @zqiao11 has submitted a PR #189. Let me know if you have any further suggest for better fine tuning guide line. If you face any issue with your fine-tuning data and evaluation data. You can share it here.

**Comment by chenghaoliu89** [comment]
Hi @liyujia011025 we dont have bandwidth for improving specific benchmark. You can follow the current example and try your data. If you have any tech issue, you can share it and we are happy to help.

**Reply by MoradLaglil** [reply]
Hi,

I would like to fine-tune Moirai2 on the GIFT-Eval Train dataset, but I couldn’t find any finetuning code in the Moirai2 repository. Could you please advise me on how to do this, or let me know if you intend to add this feature in the future? And how to finetune Moirai on Gift-eval Test (which steps i should follow)?

Thanks
